listenbrainz/webserver/static/js/src/UserPageHeading.tsx

```diff
@@ -24,29 +24,35 @@ import * as Sentry from "@sentry/react";
 import FollowButton from "./FollowButton";
 import APIService from "./APIService";
 import GlobalAppContext, { GlobalAppContextT } from "./GlobalAppContext";
+import ReportUserButton from "./ReportUser";
 import { getPageProps } from "./utils";
 
 const UserPageHeading = ({
   user,
   loggedInUser,
   loggedInUserFollowsUser = false,
+  alreadyReportedUser = false,
 }: {
   user: ListenBrainzUser;
   loggedInUser: ListenBrainzUser | null;
   loggedInUserFollowsUser: boolean;
+  alreadyReportedUser: boolean;
 }) => {
   return (
-    <h2 className="page-title">
-      {user.name}
-      {loggedInUser && user.name !== loggedInUser.name && (
-        <FollowButton
-          type="icon-only"
-          user={user}
-          loggedInUser={loggedInUser}
-          loggedInUserFollowsUser={loggedInUserFollowsUser}
-        />
-      )}
-    </h2>
+    <>
+      <h2 className="page-title">
+        {user.name}
+        {loggedInUser && user.name !== loggedInUser.name && (
+          <FollowButton
+            type="icon-only"
+            user={user}
+            loggedInUser={loggedInUser}
+            loggedInUserFollowsUser={loggedInUserFollowsUser}
+          />
+        )}
+      </h2>
+      <ReportUserButton user={user} alreadyReported={alreadyReportedUser} />
```

Comment:
Why do we not check whether the user is logged in before showing the Report button as we do for Follow button? (I assume we do not want to show the report button to not logged in users.)

Reply:
No reason, I couldn't make up my mind :) But you're right, we just shouldn't show it if the user isn't logged in, all the more because we add another component (the modal) that would be pretty much useless.